designs/Verify-Verbosity.md

```diff
@@ -0,0 +1,302 @@
+trace:       SHA256 hash: 51044706BD237B91B89B781337E6D62656C69F0FCFFBE8E43741367948127862
+trace:       Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
+trace:       Valid from: 10/22/2013 5:00:00 AM to 10/22/2028 5:00:00 AM
+trace: 
+trace:             Subject Name: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
+trace:             SHA1 hash: 0563B8630D62D75ABBC8AB1E4BDFB5A899B24D43
+trace:             SHA256 hash: 3E9099B5015E8F486C00BCEA9D111EE721FABA355A89BCF1DF69561E3DC6325C
+trace:             Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
+trace:             Valid from: 11/9/2006 4:00:00 PM to 11/9/2031 4:00:00 PM
+trace: 
+Timestamp: 12/9/2020 2:32:12 PM
+Verifying repository countersignature's timestamp with timestamping service certificate: 
+  Subject Name: CN=Symantec SHA256 TimeStamping Signer - G3, OU=Symantec Trust Network, O=Symantec Corporation, C=US
+  SHA1 hash: A9A4121063D71D48E8529A4681DE803E3E7954B0
+  SHA256 hash: C474CE76007D02394E0DA5E4DE7C14C680F9E282013CFEF653EF5DB71FDF61F8
+  Issued by: CN=Symantec SHA256 TimeStamping CA, OU=Symantec Trust Network, O=Symantec Corporation, C=US
+  Valid from: 12/22/2017 4:00:00 PM to 3/22/2029 4:59:59 PM
+trace:       Subject Name: CN=Symantec SHA256 TimeStamping CA, OU=Symantec Trust Network, O=Symantec Corporation, C=US
+trace:       SHA1 hash: 6FC9EDB5E00AB64151C1CDFCAC74AD2C7B7E3BE4
+trace:       SHA256 hash: F3516DDCC8AFC808788BD8B0E840BDA2B5E23C6244252CA3000BB6C87170402A
+trace:       Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:       Valid from: 1/11/2016 4:00:00 PM to 1/11/2031 3:59:59 PM
+trace: 
+trace:             Subject Name: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:             SHA1 hash: 3679CA35668772304D30A5FB873B0FA77BB70D54
+trace:             SHA256 hash: 2399561127A57125DE8CEFEA610DDF2FA078B5C8067F4E828290BFB860E84B3C
+trace:             Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
+trace:             Valid from: 4/1/2008 5:00:00 PM to 12/1/2037 3:59:59 PM
+
+Successfully verified package 'NuGet.Common.5.9.0-preview.2'.
+```
+
+</details>
+
+#### `--verbosity diagnostic`
+
+* Same output as `--verbosity detailed`
+
+### Example output in failure scenarios
+
+* Errors and Warnings are displayed to the console irrespective of `verbosity` level.
+
+#### `Verifying tampered package`
+
+<details>
+<summary>output</summary>
+
+```
+dotnet nuget verify tampered.12.0.1.nupkg -v -detailed
+error: NU3008: The package integrity check failed.
+error: Package signature validation failed.
+```
```

Comment:
Many folks on the Debian issue were confused due to the lack of information included in NuGet's error messages. Most folks thought the issue was a bug in NuGet when in reality the problem was due to Debian not trusting a certificate. Can we add more information in this error message?

For example:

```suggestion
```
dotnet nuget verify tampered.12.0.1.nupkg -v -detailed
error: NU3008: The package integrity check failed. Please ensure that the package has not been tampered since signing.
error: Package signature validation failed.
```

Reply:
Added `Please ensure that the package has not been tampered since signing. ` to the error message.

Reply:
While Loic's comment was based on chain building error messages --- which can be difficult to interpret ---, his comment does apply generally.

In this case, NU3008 is not a chain building warning or error.  NU3008 indicates that the package has been tampered with since being signed, which seems like a relatively straightforward error when compared with [NU3028](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu3028).

The sentence you've added has a few problems:
1. `Please` should be avoided in error messages.  See https://github.com/NuGet/Home/issues/8141.
2. The new sentence tells a user to ensure that a package has not been tampered since signing, but [_that is a role of the signature --- to provide an integrity guarantee._](https://en.wikipedia.org/wiki/Digital_signature#Integrity)  How else would a typical user ensure this?

If you feel that the current error message is not helpful enough, you could add the sentence:  `The package has been tampered with since being signed.`